src/main/java/com/ait/lienzo/client/core/shape/wires/WiresConnector.java

```diff
@@ -731,6 +663,260 @@ public Point2DArray getControlPoints(){
+                    Point2D m = new Point2D(x0,
+                                            y0);
+                    if (i == 0) {
+                        // this is position is needed, if we close the path.
+                        pathStart = m;
+                    }
+                    segmentStart = m;
+                    break;
+                }
+                case PathPartEntryJSO.LINETO_ABSOLUTE: {
+                    points = entry.getPoints();
+                    double x0 = points.get(0) + offsetX;
+                    double y0 = points.get(1) + offsetY;
+                    Point2D end = new Point2D(x0,
+                                              y0);
+
+                    if (oldPoints.get(pointsIndex).equals(segmentStart)) {
+                        pointsIndex++;
+                        color = MagnetManager.m_c_rotor.next();
+                        colorMap.put(color,
+                                     pointsIndex);
+                    }
+                    ctx.setStrokeColor(color);
+
+                    ctx.beginPath();
+                    ctx.moveTo(segmentStart.getX(),
+                               segmentStart.getY());
+                    ctx.lineTo(x0,
+                               y0);
+                    ctx.stroke();
+                    segmentStart = end;
+                    break;
+                }
+                case PathPartEntryJSO.CLOSE_PATH_PART: {
+                    double x0 = pathStart.getX() + offsetX;
+                    double y0 = pathStart.getY() + offsetY;
+                    Point2D end = new Point2D(x0,
+                                              y0);
+                    if (oldPoints.get(pointsIndex).equals(segmentStart)) {
+                        pointsIndex++;
+                        color = MagnetManager.m_c_rotor.next();
+                        colorMap.put(color,
+                                     pointsIndex);
+                    }
+                    ctx.setStrokeColor(color);
+                    ctx.beginPath();
+                    ctx.moveTo(segmentStart.getX(),
+                               segmentStart.getY());
+                    ctx.lineTo(x0,
+                               y0);
+                    ctx.stroke();
+                    segmentStart = end;
+                    break;
+                }
+                case PathPartEntryJSO.CANVAS_ARCTO_ABSOLUTE: {
+                    points = entry.getPoints();
+
+                    double x0 = points.get(0) + offsetX;
+                    double y0 = points.get(1) + offsetY;
+                    Point2D p0 = new Point2D(x0,
+                                             y0);
+
+                    double x1 = points.get(2) + offsetX;
+                    double y1 = points.get(3) + offsetY;
+                    double r = points.get(4);
+                    Point2D p1 = new Point2D(x1,
+                                             y1);
+                    Point2D end = p1;
+
```

Comment:
`p1` and `end` are useless variables. The only use is on the end of block: `segmentStart = end;`. I guess it can be just: `segmentStart = new Point(x1, y1);`